src/lists/index.ts

```diff
@@ -0,0 +1,35 @@
+import type {Riwayas} from "./types"
+import {HafsLists} from "./HafsLists"
+import { QalunLists } from "./QalunLists";
+
+
+export const riwayahs: Riwayas = {
+Hafs : HafsLists,
+Qalun : QalunLists
+}as const
+
+export function getListsOfRiwaya<R extends keyof Riwayas>(riwaya: R): Riwayas[R] {
+  return riwayahs[riwaya];
+}
+
+export const getList = <
+  R extends keyof Riwayas,
+  L extends keyof Riwayas[R]
+>(
+  listName: L,
+  riwaya?: R
+): Riwayas[R][L] => {
+  if (riwaya) {
+    return riwayahs[riwaya][listName]
+  }
+
+  // if no riwaya is provided, fallback (choose the first one that has this list)
```

Comment:
we can't return inderministic results. if no riwayah value provided, we should assume its `hafs`